app/Controllers/Event.php

```diff
@@ -75,4 +75,51 @@ public function get(int|null $year = null) {
             'talks' => $talks,
         ]);
     }
+
+    public function getICalendarFile(int $year) {
+        // Start output buffering
+        ob_start();
```

Comment:
Do you need the output buffer at all? It seems like you are collecting the "output" into the `$icsContent` variable.

Reply:
Honestly: I don't know. I took @freddylinux' code and just left this in. If you think that it doesn't make a difference, I can also remove it.

Reply:
From the [documentation](https://www.php.net/manual/en/function.ob-start.php) I read that it is capturing the "script output" aka "echo". And you do not use any of these in that function.

So I would remove it and the `ob_end_clean()` as well.

Reply:
I have no opinion on that.

Reply:
> From the [documentation](https://www.php.net/manual/en/function.ob-start.php) I read that it is capturing the "script output" aka "echo". And you do not use any of these in that function.
> 
> So I would remove it and the `ob_end_clean()` as well.

Sounds reasonable, will remove it 👍